app/src/main/kotlin/com/strv/archdemo/ui/counter/CounterViewModel.kt

```diff
@@ -1,19 +1,40 @@
 package com.strv.archdemo.ui.counter
 
 import android.os.Parcelable
+import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.ViewModel
 import com.strv.archdemo.extension.toBundle
+import com.strv.archdemo.ktools.Preferences
+import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.parcelize.Parcelize
+import javax.inject.Inject
 
 @Parcelize
 data class CounterConfig(val text: String) : Parcelable
 
-class CounterViewModel(
-    val savedStateHandle: SavedStateHandle
+@HiltViewModel
+class CounterViewModel @Inject constructor(
+    savedStateHandle: SavedStateHandle,
+    val preferences: Preferences,
 ) : ViewModel() {
     private val args = CounterFragmentArgs.fromBundle(savedStateHandle.toBundle())
 
     // We should always hold all data (state) inside ViewModel
     val title = args.config.text
+    val clickCount = MutableLiveData(preferences.getCounterValueFromPreferences())
+
+    internal fun onMinusClick() {
+        val currentClickValue = clickCount.value ?: 0
```

Comment:
Can you rather make the whole `clickCount` non nullable by doing

```
MutableLiveData(preferences.getCounterValueFromPreferences() ?: 0)
```

Just so that you don't have to do the check at every place you touch it.